src/state_machine.rs

```diff
@@ -379,23 +379,29 @@ impl StateMachine {
     }
 
     pub fn pause(self) -> Result<PausedState, RoutingError> {
-        // TODO: should we allow pausing from other states too?
-        match self.state {
-            State::Elder(state) => {
-                let mut state = state.pause()?;
-                state.network_rx = Some(self.network_rx);
-                Ok(state)
-            }
-            _ => Err(RoutingError::InvalidStateForOperation),
-        }
+        info!("{} - Pause", self.current());
+
+        let mut paused_state = match self.state {
+            State::Elder(state) => state.pause()?,
```

Comment:
can we have something like `state.pause(self.network_rx)` so that we can avoid assigning it separately?

Reply:
(Not sure that would be better as it would require the state to know about that detail.)

Reply:
it has to, as there is assignment later on immediately ?

Reply:
neither elder nor adults do

Reply:
https://github.com/maidsafe/routing/pull/1936/files#diff-e79e9ddf41b90c06a8194192f55e31deR390 ?

Reply:
Yeah, I wouldn't worry too much about details like this. This pause/resume mechanics is not finished yet and there are likely going to be changes, both the the internal details and possibly to the public API as well. There is issue raised for it: https://github.com/maidsafe/routing/issues/1937